### PR TITLE
Classifier: Refactor LightCurveViewer to use Chart component

### LINKED ISSUE #3654: With Safari v16 light curve viewer height expands unexpectedly
## Package
- lib-classifier, [LightCurveViewer](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js#L427)

## Describe the bug
Introduced in Safari v16, and only Safari v16, the light curve viewer height expands unexpectedly. 
Related to #3119 , #3139 , and #3646 .

## To Reproduce
1. In Safari v16, go to https://www.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess/classify/workflow/11235?demo=true
2. Note the light curve subject viewer height continues to expand, making classifier unusable.

Staging project local links:
- lib-classifier - https://local.zooniverse.org:8080/?project=1862&workflow=3317
- app-project - https://local.zooniverse.org:3000/projects/nora-dot-test/planet-finders-beta/classify/workflow/3317

## Expected behavior
The light curve viewer height is reasonably large, depending on browser sizing about 400-500px

### Desktop (please complete the following information):
 - OS: macOS Monterey 12.5.1
 - Browser: Safari
 - Version: 16

### PR DESCRIPTION
**Pull request by eatyourgreens**
- add named exports to `@viewers/components/SVGComponents`.
- Use the `Chart` component as the root `svg` element of `LightCurveViewer`.
- refactor `LightCurveViewer` to use `height` and `width` from the SVG's client bounding rectangle.
- draw the SVG chart on component mount.
- wrap `LightCurveViewer` in a fixed-height box, like `ScatterPlotViewer`.

## Package
lib-classifier

- Closes #3654.

## How to Review
Dev classifier: https://localhost:8080/?project=nora-dot-eisner/planet-hunters-tess&env=production&demo=true

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js [hunk 1]
```diff
@@ -128,8 +125,16 @@ class LightCurveViewer extends Component {
...
+      dataExtent,
+      dataPoints,
+      innerMargin,
+      outerMargin
+    } = this.props
+    const container = this.svgContainer.current
+    const { height, width } = container?.getBoundingClientRect()
     if (height && width) {
       this.chartWidth = width
       this.chartHeight = height
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js:140]: This doesn’t seem to be used anywhere.

Comment (shaunanoordin) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js:140]: I agree, `chartWidth` and `chartHeight` appear unused. I introduced them in #385, and these might have been leftovers from an earlier WIP, or possibly some resizing feature that didn't pan out.

There's another part of this code that sets `this.chartWidth = 100` and `this.chartHeight = 100` that's similarly unused.

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Package: `lib-classifier`
Affects: Light Curve Viewer (and therefore [Planet Hunters TESS](https://www.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess))
Fixes #3654, which in turn is a sorta-echo of #3139

This PR refactors the LCV subject viewer and fixes an issue where the LCV's height increased uncontrollably (to the point the subject viewer becomes unusable) on Safari 16.

- 🗝️ Key factor here seems to be the addition of a fixed-height `<Box>` in `LightCurveViewerContainer.js`
- Observations on the code refactor:
  - Instead of dynamically constructing the `<svg>` element at `LightCurveViewer.initChart()`, the svg element is taken from the `Chart` component. It's otherwise the same, DOM-wise
  - Note that the `<Chart>` component has height=100% and width=100% _by default,_ (Chart.defaultProps) so the `<svg>` gets those values too.

### Dev Notes

The removal of `react-resize-detector` (RRD) introduces a new quirk: the size of the chart is now _only_ set when a new Subject loads in. If the volunteer resizes their browser window in the middle of classifying, the size of the chart remains unchanged, either causing parts of the data to be cut off (because the container around the SVG is now too small) or "blank spaces" to appear (because the container around the SVG is now too large).

https://user-images.githubusercontent.com/13952701/190431571-de8ef02e-0820-448a-9cfd-377db06cbaa6.mov

The workaround is to either reload the page, proceed to the next subject, or resize the browser window to the original size.

If RRD is contributing to the sizing issues of the LCV on Safari, or is otherwise outdated, then it's worth removing.

### Testing

Tested with macOS, on localhost, targeting [Planet Hunters TESS on lib-classifier](https://local.zooniverse.org:8080/?env=production&project=nora-dot-eisner/planet-hunters-tess), with the following browsers: **Safari 16.0**, FF 104, Chrome 105, and **Safari 15.3** (via Browserstack)

- I'm able to view the Light Curve Viewer. The LCV appears comfortably visible within a desktop-sized browser window (around 1000px x 666px), and doesn't extend beyond the viewport. I can click on the viewer without having to scroll the page, etc 
    image
- I can zoom in/out and pan the subject "image" via mouse, keyboard, and (for zoom only) the UI buttons.
- I can place annotations, edit annotations, and delete annotations. (Using mouse.)

### Status

Changes LGTM. This PR fixes the major (breaking) display issue in Safari 16 (and maintains parity for Safari 15, Chrome, & FF), though please note the new quirk introduced by the removal of RRD.

**Comment by goplayoutside3** [comment]
> The removal of react-resize-detector (RRD) introduces a new quirk: the size of the chart is now only set when a new Subject loads in. If the volunteer resizes their browser window in the middle of classifying, the size of the chart remains unchanged

Is there ever a case where this volunteer behavior could cause an incorrect annotation submitted during classification?

**Pull request merged**